src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -406,6 +411,36 @@ private SmugglerProgressBase.Counts ProcessDocuments(SmugglerResult result, Buil
             return result.Documents;
         }
 
+        private void WriteAttachments(Document document, IDocumentActions actions)
+        {
+            var source = _source as DatabaseSource;
+            var streamDestination = actions as StreamDestination.StreamDocumentActions;
+            if (source == null || streamDestination == null)
+                return;
+
+            if ((document.Flags & DocumentFlags.HasAttachments) != DocumentFlags.HasAttachments || 
+                document.Data.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) == false || 
+                metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray attachments) == false)
+                return;
+
+            if (attachmentStreamsAlreadyExported == null)
+                attachmentStreamsAlreadyExported = new HashSet<LazyStringValue>();
```

Comment:
Is it valid to save the LazyStringValue here? What context is it attached to, and are you sure it wouldn't be reset midway?